src/mapIndexed.js

```diff
@@ -0,0 +1,39 @@
+import { addIndex, map } from 'ramda';
+
+
+/**
+ * Like {@link http://ramdajs.com/docs/#map|R.map} but with iterator function that take the current element,
+ * the element's key, and the list itself.
+ *
+ * This function is like {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map|Array.prototype.map}
+ *
+ * `mapIndexed` implementation is simple : `
+ * const mapIndexed = R.addIndex(R.map);
+ * `
+ * @func mapIndexed
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.5.0|v2.5.0}
+ * @category List
+ * @sig Functor f => ((a, Number, f a) => b) => f a -> f b
+ * @param {Function} fn The iterator function.
+ * Receives tree values, the current element, the element's key and the list itself
+ * @param {Array} list The list to be iterated over.
+ * @return {Array} The new list.
+ * @see {@link http://ramdajs.com/docs/#addIndex|R.addIndex}, {@link http://ramdajs.com/docs/#map|R.map}.
+ * @example const joinArgs = separator => R.compose(R.join(separator), RA.list);
```

Comment:
@example is too complex. Replace with the one that [R.addIndex](http://ramdajs.com/docs/#addIndex) uses.
```js
R.mapIndexed((val, idx, list) => idx + '-' + val, ['f', 'o', 'o', 'b', 'a', 'r']);
//=> ['0-f', '1-o', '2-o', '3-b', '4-a', '5-r']